packages/delegation-toolkit/src/userOp.ts

```diff
@@ -266,50 +259,93 @@ export const SIGNABLE_USER_OP_TYPED_DATA: TypedData = {
- * @param params.signer - The signatory to use for signing.
- * @param params.userOperation - The user operation to sign.
+ * Prepares typed data for user operation signing.
+ * This is an internal helper function that's not exposed in the public API.
+ *
+ * @param params - The parameters for preparing the typed data.
+ * @param params.userOperation - The user operation to prepare for signing.
  * @param params.entryPoint - The entry point contract address.
+ * @param params.entryPoint.address - The address of the entry point contract.
  * @param params.chainId - The chain ID that the entry point is deployed on.
  * @param params.name - The name of the domain of the implementation contract.
  * @param params.version - The version of the domain of the implementation contract.
  * @param params.address - The address of the smart account.
- * @param params.entryPoint.address - The address of the entry point contract.
- * @returns The signature of the user operation.
+ * @returns The prepared typed data for signing.
  */
-export const signUserOperation = async ({
-  signer,
+export const prepareSignUserOperationTypedData = ({
   userOperation,
   entryPoint,
   chainId,
   name,
   address,
   version = '1',
 }: {
-  signer: WalletClient<Transport, Chain, Account>;
   userOperation: Omit<UserOperationV07, 'signature'>;
   entryPoint: { address: Address };
   chainId: number;
-  address: Address;
   name: 'HybridDeleGator' | 'MultiSigDeleGator';
+  address: Address;
   version?: string;
 }) => {
   const packedUserOp = toPackedUserOperation({
     ...userOperation,
     signature: '0x',
   });
 
-  return signer.signTypedData({
-    account: signer.account,
+  return {
     domain: {
       chainId,
       name,
       version,
       verifyingContract: address,
     },
     types: SIGNABLE_USER_OP_TYPED_DATA,
-    primaryType: 'PackedUserOperation',
+    primaryType: 'PackedUserOperation' as const,
     message: { ...packedUserOp, entryPoint: entryPoint.address },
+  };
+};
+
+/**
+ * Signs a user operation using a private key.
+ * @param params - The parameters for signing the user operation.
+ * @param params.privateKey - The private key to use for signing.
+ * @param params.userOperation - The user operation to sign.
+ * @param params.entryPoint - The entry point contract address.
+ * @param params.chainId - The chain ID that the entry point is deployed on.
+ * @param params.name - The name of the domain of the implementation contract.
+ * @param params.version - The version of the domain of the implementation contract.
+ * @param params.address - The address of the smart account.
+ * @param params.entryPoint.address - The address of the entry point contract.
+ * @returns The signature of the user operation.
+ */
+export const signUserOperation = async ({
+  privateKey,
+  userOperation,
+  entryPoint,
+  chainId,
+  name,
+  address,
+  version = '1',
+}: {
+  privateKey: `0x${string}`;
```

Comment:
Can we make this type `Hex`?

We actually do this quite a lot throughout the codebase. Maybe we can leave this if there's no other changes needed, and just create a ticket to go through and change all of the instances.